app/src/matrix_transform.c

```diff
@@ -45,21 +80,21 @@ int32_t zmk_matrix_transform_row_column_to_position(uint32_t row, uint32_t colum
     row += DT_PROP(ZMK_KEYMAP_TRANSFORM_NODE, row_offset);
 #endif
 
-    const uint32_t matrix_index = (row * ZMK_MATRIX_COLS) + column;
-
 #ifdef ZMK_KEYMAP_TRANSFORM_NODE
-    if (matrix_index >= ARRAY_SIZE(transform)) {
-        return -EINVAL;
-    }
 
-    const uint32_t value = transform[matrix_index];
+    for (int i = 0; i < mt->entries_len; i++) {
+        const struct zmk_matrix_transform_entry *entry = &mt->entries[i];
 
-    if (!value) {
-        return -EINVAL;
+        if (entry->row == row && entry->column == column) {
+            return i;
+        }
```

Comment:
Would it be worth constructing a reverse lookup table for each transform (or just for the dynamically chosen one at the time of choosing?) and using that for constant time lookups?

I am not sure how often this function is called, but if it is called on every key position change then it would add up.

Reply:
Oh, it looks like the `transform` in the old code was doing that? It does require knowing the max row/col values which isn't ideal.

Reply:
Yeah. As I've slept on this more, maybe we:

* Use the max x/y values set on the *transform*
* Use that info to easily create that constant lookup table for a given transform.
* Stop inferring those max x/y from properties on the kscan node, which is a major headache and forces a dependency on kscan driver internals. Painful to use third party kscan like my EC analog driver.

Thoughts?

Reply:
Yeah, sounds very reasonable to me. Setting `rows`/`columns` properties on matrix transforms + kscans (including composite ones) were always a bit confusing IMO, even for me. Do I understand correctly that none of those will be necessary in this case?

Reply:
With this change, it would be required consistently *only* on the matrix transform.

Reply:
Can that not be inferred from the item values in the `map`?

Reply:
We could at *runtime* if we want to, but not in our preprocessor to create it statically.

Reply:
Not necessarily, but it *does* force it to be non-const, so it'd be in RAM not ROM.

Reply:
Ok, figured out how to restore the constant time lookup table for all scenarios, and ditched the "entries" approach I'd used before.

Also took the time to fix up the col/row offsets so this should still work fine for splits.

Reply:
Great, FWIW it looks good to me!